packages/openapi-types/index.ts

```diff
@@ -182,7 +188,8 @@ export namespace OpenAPIV3_1 {
     }
   >;
 
-  export type DiscriminatorObject = OpenAPIV3.DiscriminatorObject;
+  export type DiscriminatorObject = OpenAPIV3.DiscriminatorObject &
```

Comment:
This is extensible in 3.1, but not in 3.0 (or at least the specification doesn't explicitly call this out).